tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Steps/UserSteps.cs

```diff
@@ -7,29 +18,135 @@ namespace NHSD.BuyingCatalogue.Identity.Api.IntegrationTests.Steps
     [Binding]
     internal sealed class UserSteps
     {
+        private const string OrganisationMapDictionary = "OrganisationMapDictionary";
+        private const string AccessTokenKey = "AccessToken";
+        private readonly string _organisationUrl;
+
         private readonly ScenarioContext _context;
+        private Response _response;
+        private Settings _settings;
 
-        public UserSteps(ScenarioContext context)
+        public UserSteps(ScenarioContext context, Response response, Settings settings)
         {
             _context = context;
+            _response = response;
+            _settings = settings;
+            _organisationUrl = settings.OrganisationApiBaseUrl + "/api/v1/Organisations";
         }
 
-        [Given(@"There are Users in the database")]
-        public void GivenThereAreUsersInTheDatabase(Table table)
+        [Given(@"Users exist")]
+        public async Task GivenUsersExist(Table table)
         {
-            var users = table.CreateSet<UserTable>();
+            var users = table.CreateSet<NewUserTable>();
+            foreach (var user in users)
+            {
+                var allOrganisations = _context.Get<IDictionary<string, Guid>>(OrganisationMapDictionary);
+
+                var organisationId = Guid.Empty;
+                if (allOrganisations.ContainsKey(user.OrganisationName))
+                {
+                    organisationId = allOrganisations[user.OrganisationName];
+                }
+
+                var userEntity = new UserEntity
+                    {
+                        PasswordHash = GenerateHash(user.Password),
```

Comment:
I may be out of my depth here, but are we gaining any value by implementing a cryptographic hash function for populating test passwords? 
If the structure of the password is important to test, should there be tests just for that perhaps?

Reply:
the value is that the generated users can be used for logging in, which makes our tests that log in be independent from the test data that's injected when we spin up the environment

Reply:
I see, wouldn't a pseudo-randomly generated string of characters suffice for this?

Reply:
unfortunately not, there's very specific requirements when it comes to the password hash column in identity server

Reply:
Now I undestand. Thank you for clarifying.